phpstan-baseline.neon

```diff
@@ -0,0 +1,127 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Parameter \\#1 \\$children of method Knp\\\\Menu\\\\ItemInterface\\:\\:setChildren\\(\\) expects array\\<int\\|string, Knp\\\\Menu\\\\ItemInterface\\>, array\\<Knp\\\\Menu\\\\ItemInterface\\>\\|false given\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Possibly invalid array key type Knp\\\\Menu\\\\ItemInterface\\|string\\.$#"
+			count: 3
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Binary operation \"\\.\" between 'Cannot find…' and Knp\\\\Menu\\\\ItemInterface\\|string results in an error\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Method Knp\\\\Menu\\\\MenuItem\\:\\:getFirstChild\\(\\) should return Knp\\\\Menu\\\\ItemInterface but returns Knp\\\\Menu\\\\ItemInterface\\|false\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Method Knp\\\\Menu\\\\MenuItem\\:\\:getLastChild\\(\\) should return Knp\\\\Menu\\\\ItemInterface but returns Knp\\\\Menu\\\\ItemInterface\\|false\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Cannot call method getLastChild\\(\\) on Knp\\\\Menu\\\\ItemInterface\\|null\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Cannot call method getFirstChild\\(\\) on Knp\\\\Menu\\\\ItemInterface\\|null\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Cannot call method getChildren\\(\\) on Knp\\\\Menu\\\\ItemInterface\\|null\\.$#"
+			count: 2
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Method Knp\\\\Menu\\\\MenuItem\\:\\:getIterator\\(\\) should return Traversable\\<string, Knp\\\\Menu\\\\ItemInterface\\> but returns ArrayIterator\\<int\\|string, Knp\\\\Menu\\\\ItemInterface\\>\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Parameter \\#1 \\$name of method Knp\\\\Menu\\\\MenuItem\\:\\:getChild\\(\\) expects string, int\\|string given\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Parameter \\#1 \\$name of method Knp\\\\Menu\\\\MenuItem\\:\\:removeChild\\(\\) expects Knp\\\\Menu\\\\ItemInterface\\|string, int\\|string given\\.$#"
+			count: 1
+			path: src/Knp/Menu/MenuItem.php
+
+		-
+			message: "#^Parameter \\#1 \\$value of method Knp\\\\Menu\\\\Renderer\\\\Renderer\\:\\:escape\\(\\) expects string, string\\|null given\\.$#"
+			count: 2
+			path: src/Knp/Menu/Renderer/ListRenderer.php
+
+		-
+			message: "#^Method Knp\\\\Menu\\\\Renderer\\\\ListRenderer\\:\\:renderLabel\\(\\) should return string but returns string\\|null\\.$#"
```

Comment:
I think this is caused by a bad phpdoc on `ItemInterface::getLabel()`. The phpdoc says it defaults to the item name, so I don't think it can actually be null.

Reply:
Do you think we can change interface to return `string` instead of `?string`?

Reply:
I think using a nullable typehint was a mistake when types were introduced. the phpdoc was saying `string` before that: https://github.com/KnpLabs/KnpMenu/commit/2dd2e7687d23b174689c357cd4a749ff825e4795#diff-8f6d2b955ef3caa9f7aa107c36571bad58fc424afbbc1abdbb4e5a4588d63634R56

Reply:
I agree. Do you think we can change that return type in a minor version, adding a note in changelog?

Reply:
given that MenuItem never returns `null` there, I think we can. I don't expect people to have custom implementations of the ItemInterface actually.